README.md

```diff
@@ -16,48 +16,62 @@ repository is compatible with the official Fast Downward Git repository.
   resulting Git repository will be written to. The optional parameter can be used to 
   redirect the output of fast-export to a file.
 
-    ./run-cleanup-and-conversion.sh MERCURIAL_REPOSITORY CONVERTED_GIT_REPOSITORY \
+    ./run-all-steps.sh MERCURIAL_REPOSITORY CONVERTED_GIT_REPOSITORY \
         [--redirect-fast-export-stderr FILE]
 
   The conversion is done in two steps that can also be run individually. In this case
-  CLEANED_MERCURIAL_REPOSITORY is a location where the intermediate cleaned up Mercurial 
-  repository will be written to:
-
-    ./run-cleanup.sh MERCURIAL_REPOSITORY CLEANED_MERCURIAL_REPOSITORY
+  ORDERED_REPOSITORY is an intermediate repository that ensures that the history
+  contains all commits in the same order as the Fast Downward master repository.
+  The CLEANED_MERCURIAL_REPOSITORY is a location where the intermediate cleaned
+  up Mercurial repository will be written to:
+  
+    ./run-cleanup.sh MERCURIAL_REPOSITORY ORDERED_REPOSITORY CLEANED_MERCURIAL_REPOSITORY
```

Comment:
I see this as 1 intermediate repository (ORDERED_REPOSITORY). CLEANED_* is the intended output of this script, but I also dislike interface. We do not see another way to solve what we want to do. We need to 1. order the commits and 2. cleanup the repository. We do NOT want to modify the MERCURIAL_REPOSITORY.
To order the commits, we have to clone hg.fast-downward.org anew. This requires a directory and pull the users changes into it. The cleanup (hg convert) cannot be done inplace. Thus, we have to create a second directory (we do not overwrite the users repository).

Executing this script (instead of the run-all-steps.sh) shall give the user the ability to investigate what could have gone wrong. Thus, I do not create a temporary directory and delete at the end of the cleanup.

Reply:
That all sounds like an argument for having two separate steps for ordering and cleanup. What was the reasoning merging the steps? If the point was that the ordering is something internal, then we can make it more internal by using a temporary directory for it and not bothering the user with this. But if we want to keep the ordered repository after the script runs, it sounds like we should have two steps again.

Reply:
The ordering is more an internal thing for the cleanup. We observed the bad design, when Jendrik ran the ordering which took some time and then the cleanup failed because its requirement (all changes pulled in) were not satisfied. The user should see this directly. Adding the check that all changes are pulled in to the run-all-steps.sh script would be a hack. Instead we made the ordering internal to the cleanup. 

We could make the ordered repository a temporary one. For me this script is used by a user who wants to have full control. Therefore, I kept the ordered repository. The user can investigate into it or just add `rm DIR` to his command. If the majority wants it to be temporary and automatically removed, we can change this.

Reply:
I don't know, it still sounds like this should be separate steps to me. If there are conditions that the input (`MERCURIAL_REPOSITORY`) has to satisfy, they should be check at the start of the first step. If the cloning takes too long to check if there are incoming changes, you could do `hg -R MERCURIAL_REPOSITORY incoming hg.fast-downward.org` instead . But again, if you already discussed this, I don't want to restart that discussion.

Reply:
I also disagree with keeping an intermediate result around, and even making it a command-line argument. Like Florian said it breaks the abstraction. If we want to facilitate debugging if things go wrong, for me the logical thing is simply to not go out of our way and delete intermediate results in cases where things go wrong, but they should not be kept around if things go right.

I also agree it makes no sense to first clone and *then* test and fail if there are incoming commits. The test about incoming commits should be made right at the start if we consider it a reason to abort.

We also agreed on these points in the discussion with Silvan and Jendrik.

For what it's worth, regarding debugging: unless there are network errors or similar basic issues, there is no way in which the initial clone-and-pull steps could go wrong.

If you're OK with it, I'll push a suggested change -- if you don't like it, we can strip it or back it out.